plugins/modules/ovirt_disk.py

```diff
@@ -716,6 +719,36 @@ def update_storage_domains(self, disk_id):
 
         return changed
 
+    def update_disk_profile(self, disk_id):
+        """
+        Updates the disk_profile for a given disk
+
+        :param disk_id: id of the disk
+        :return: Wether a change has been made to the disk or not
+        """
+        changed = False
+        disk_service = self._service.service(disk_id)
+        disk = disk_service.get()
+        dp_service = self._connection.system_service().disk_profiles_service()
+
+        if self._module.params['profile']:
+            new_profile_id = get_id_by_name(dp_service, self._module.params['profile'])
+            if new_profile_id == disk.disk_profile.id:
+                return changed
+            update_disk = otypes.Disk(
+                id=disk.id,
+                disk_profile=otypes.DiskProfile(id=new_profile_id),
+            )
+            changed = self.action(
+                action='update',
+                disk=update_disk,
+                action_condition=lambda d: new_profile_id != d.disk_profile.id,
+                wait_condition=lambda d: d.status == otypes.DiskStatus.OK,
+                post_action=lambda _: time.sleep(self._module.params['poll_interval']),
+            )['changed']
+
+        return changed
```

Comment:
Is it necessary to have a separate method for an update of the disk profiles?
I think it should go to the `update_check`
https://github.com/oVirt/ovirt-ansible-collection/blob/master/plugins/modules/ovirt_disk.py#L719